src/main/kotlin/thatline/localup/etcapi/service/ForecastService.kt

```
@@ -121,4 +120,33 @@ class ForecastService(
+
+    /**
+     * 현재 시간을 기준으로 유효한 기상청 API baseDate와 baseTime을 반환합니다.
+     * 기상청 단기예보는 하루 8번 (02, 05, 08, 11, 14, 17, 20, 23시) 발표됩니다.
+     */
+    private fun getValidBaseDateAndTime(dateTime: LocalDateTime): Pair<String, String> {
+        val validTimes = listOf("0200", "0500", "0800", "1100", "1400", "1700", "2000", "2300")
+        val currentHour = dateTime.hour
+        val currentMinute = dateTime.minute
+
+        // 현재 시간을 4자리 문자열로 변환 (예: 13:30 -> "1330")
+        val currentTime = String.format("%02d%02d", currentHour, currentMinute)
```

detekt (1.23.8)

[warning] String.format("%02d%02d", currentHour, currentMinute) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)